[Issue] gfx950 - Linux Pytorch Tests with ''gfx950-dcgpu' unsupported error

### Run:
- https://github.com/ROCm/TheRock/actions/runs/19730528508

### Jobs:
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56534537852
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56534083377
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533724247
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533388382
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533546606
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533524726
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533795844
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56533950759
- https://github.com/ROCm/TheRock/actions/runs/19730528508/job/56534111391

### Error:
```

HIP_VISIBLE_DEVICES already set to: 0
Getting GPUs supported by the current PyTorch build...done
Getting visible GPUs...done
Supported AMD GPUs: ['gfx950']
Visible AMD GPUs: ['gfx950']
Error:  Requested GPU 'gfx950-dcgpu' not found in visible GPUs that are supported by PyTorch
Error: Process completed with exit code 1.
```

--- a/external-builds/pytorch/run_linux_pytorch_tests.py
+++ b/external-builds/pytorch/run_linux_pytorch_tests.py
@@ -378,9 +378,14 @@ def detect_amdgpu_family(amdgpu_family: str = "") -> list[str]:
         )
     else:
         # Mode 3: Specific GPU arch - validate it is visible and supported by the current PyTorch build.
+
+        # We have gfx1151 -> we want to match exactly gfx1151
+        # We have gfx950-dcgpu -> we need to match exactly gfx950
+        # So remove the suffix after '-'
+        pruned_amdgpu_family = amdgpu_family.split("-")[0]
         for idx, gpu in enumerate(visible_gpus):
             if gpu in supported_gpus:
-                if gpu == amdgpu_family or amdgpu_family in gpu:
+                if gpu == pruned_amdgpu_family or pruned_amdgpu_family in gpu:
                     selected_gpu_indices += [idx]
                     selected_gpu_archs += [gpu]
 

